File: quad_controller/src/pid_controller.cpp
/*

PID controller class

Author: Sean Cassero

*/


class PIDController{

private:

  float _kp; // proportional control constant
  float _kd; // derivative control parameter
  float _ki; // integral control parameter
  float _last_error; // last error measured
  float _last_timestamp; // last recorded timestamp
  float _error_sum; // sum of all state errors, used for integral control
  float _max_windup; // max windup value


public:

  PIDController(); // class constructor
  void reset(); // reset the instance
  void setTarget(float target); // specify the desired state for PID class
  void setKP(float kp){this->_kp=kp;}; // set kp
  void setKI(float ki){this->_ki=ki;};// set ki
  void setKD(float kd){this->_kd=kd;}; // set kd
  void setMaxWindup(float max_windup){this-> _max_windup = max_windup ;}; // set max windup
  void update(float measured_value, float timestamp); // update the PID class

};

void PIDController::update(float measured_value, float timestamp){}

void PIDController::reset(){}


int main() {

  return 0;
}

File: quad_controller/src/quad_controller/pid_controller.py
# Copyright (C) 2017 Electric Movement Inc.
# All Rights Reserved.

# Author: Brandon Kinman


class PIDController:
    def __init__(self, kp = 0.0, ki = 0.0, kd = 0.0, max_windup = 10):
        #TODO
        self.kp_ = kp
        self.ki_ = ki
        self.kd_ = kd

        self.start_time_ = 0.0
        self.error_sum_ = 0.0
        self.last_timestamp_ = 0.0
        self.last_error_ = 0.0
        
    def reset(self):
        #TODO
        pass

    def setTarget(self, target):
        #TODO
        pass

    def setKP(self, kp):
        #TODO
        pass

    def setKI(self, ki):
        #TODO
        pass

    def setKD(self, kd):
        #TODO
        pass

    def setMaxWindup(self, max_windup):
        #TODO
        pass

    def update(self, measured_value, timestamp):
        #TODO
        pass
